homeassistant/helpers/entity_component.py

```diff
@@ -57,13 +53,18 @@ def setup(self, config):
             self._setup_platform(p_type, p_config)
 
         if self.discovery_platforms:
-            discovery.listen(self.hass, self.discovery_platforms.keys(),
-                             self._entity_discovered)
+            discovery.listen(
+                self.hass, self.discovery_platforms.keys(),
+                lambda service, info:
+                self._setup_platform(self.discovery_platforms[service], {},
```

Comment:
I am assuming that `discovery.listen` will not trigger callbacks for services that we do not pass in.